cleverhans/attacks.py

```diff
@@ -446,6 +446,40 @@ def fgsm(x, predictions, eps, back='tf', clip_min=None, clip_max=None):
         return fgm(x, predictions, eps, clip_min=clip_min, clip_max=clip_max)
 
 
+def vatm(model, x, predictions, eps, back='tf', num_iterations=1, xi=1e-6,
+         clip_min=None, clip_max=None):
+    """
+    A wrapper for the perturbation methods used for virtual adversarial
+    training : https://arxiv.org/abs/1507.00677
+    It calls the right function, depending on the
+    user's backend.
+    :param model: the model which returns the network unnormalized logits
```

Comment:
`model` should be a keras model or function which returns the output tensor. I renamed though `predictions` to `logits` as it is necessary for vat to have unnormalized output tensor.